lib/src/chart/bar_chart/bar_chart.dart

```diff
@@ -20,18 +23,55 @@ class BarChart extends ImplicitlyAnimatedWidget {
     Duration duration = const Duration(milliseconds: 150),
     @Deprecated('Please use [curve] instead') Curve? swapAnimationCurve,
     Curve curve = Curves.linear,
-  }) : super(
+    this.transformationController,
+    this.scaleAxis = ScaleAxis.none,
+    this.maxScale = 2.5,
+    this.minScale = 1,
+    this.trackpadScrollCausesScale = false,
+  })  : assert(
+          switch (data.alignment) {
+            BarChartAlignment.center ||
+            BarChartAlignment.end ||
+            BarChartAlignment.start =>
+              scaleAxis != ScaleAxis.horizontal && scaleAxis != ScaleAxis.free,
+            _ => true,
+          },
+          'Can not scale horizontally when BarChartAlignment is center, '
+          'end or start',
+        ),
+        super(
           duration: swapAnimationDuration ?? duration,
           curve: swapAnimationCurve ?? curve,
         );
 
   /// Determines how the [BarChart] should be look like.
   final BarChartData data;
 
+  /// The transformation controller to control the transformation of the chart.
+  final TransformationController? transformationController;
```

Comment:
Does it make sense to change the name to `chartScaleController` or `scaleTransformationController`?
I mean I think we should mention the `scale` here because it will be misleading for those who don't want to use the scale feature at all.

Reply:
If we add the `scale` keyword, then the users can map it to the `minScale` and `maxScale` properties that we have.

Reply:
And now I see the complexity that you talked about.
It's about the Matrix4 calculation.

Is it a good idea to create a wrapper around the `TransformationController` called `FlTransformationController` that just exposes the scale property?